src/v2/Apps/NewForYou/Components/NewForYouArtworksGrid.tsx

```diff
@@ -1,117 +1,121 @@
-import { Box, Button, Spacer, Text } from "@artsy/palette"
-import React, { FC, Fragment } from "react"
-import {
-  createPaginationContainer,
-  graphql,
-  RelayPaginationProp,
-} from "react-relay"
-import ArtworkGridItemFragmentContainer from "v2/Components/Artwork/GridItem"
-import { Masonry } from "v2/Components/Masonry"
-import { extractNodes } from "v2/Utils/extractNodes"
+import { Text } from "@artsy/palette"
+import React, { FC, useCallback, useState } from "react"
+import { createRefetchContainer, graphql, RelayRefetchProp } from "react-relay"
+import { PaginationFragmentContainer } from "v2/Components/Pagination"
+import createLogger from "v2/Utils/logger"
 import { NewForYouArtworksGrid_viewer } from "v2/__generated__/NewForYouArtworksGrid_viewer.graphql"
+import ArtworksGrid from "v2/Components/ArtworkGrid"
 
+const logger = createLogger("NewForYouArtworksGrid.tsx")
 interface NewForYouArtworksGridProps {
   viewer: NewForYouArtworksGrid_viewer
-  relay: RelayPaginationProp
+  relay: RelayRefetchProp
 }
 
+const PAGE_SIZE = 10
+
 export const NewForYouArtworksGrid: FC<NewForYouArtworksGridProps> = ({
   viewer,
   relay,
 }) => {
-  const [loading, setLoading] = React.useState(false)
-  const artworks = extractNodes(viewer.artworksForUser)
-
-  const loadMore = () => {
-    if (!relay.hasMore() || relay.isLoading()) return
-
-    setLoading(true)
+  const { pageInfo, pageCursors } = viewer.artworksForUser!
+  const { hasNextPage, endCursor } = pageInfo
+  const [loading, setLoading] = useState(false)
 
-    relay.loadMore(25, err => {
-      if (err) console.error(err)
+  const loadAfter = useCallback(
+    (cursor: string) => {
+      setLoading(true)
+      relay.refetch(
+        {
+          after: cursor,
+          before: null,
+          first: PAGE_SIZE,
+          last: null,
+        },
+        null,
+        error => {
+          if (error) {
+            logger.error(error)
+          }
+          setLoading(false)
+        }
+      )
+    },
+    [relay]
+  )
 
-      setTimeout(() => {
-        setLoading(false)
-      }, 2500)
-    })
-  }
+  const loadNext = useCallback(() => {
+    if (hasNextPage && endCursor) {
+      loadAfter(endCursor)
+    }
+  }, [hasNextPage, endCursor, loadAfter])
 
   return (
     <>
-      {artworks?.length > 0 ? (
-        <Masonry columnCount={[2, 3, 4]}>
-          {artworks.map(artwork => {
-            return (
-              <Fragment key={artwork.internalID}>
-                <ArtworkGridItemFragmentContainer artwork={artwork} />
-                <Spacer mt={4} />
-              </Fragment>
-            )
-          })}
-        </Masonry>
+      {viewer.artworksForUser ? (
+        <ArtworksGrid
+          artworks={viewer.artworksForUser}
+          columnCount={[2, 3]}
+          itemMargin={40}
+          loading={loading}
+        />
```

Comment:
interestingly `Masonry` can be used fine with this pagination style. It doesn't matter either way, really; though `ArtworksGrid` is much more complex.

Reply:
It is more complex, but if we've got an artwork grid it should go through the `<ArtworkGrid>` component so we can properly refactor later.